base/stl_helpers.hpp

```diff
@@ -51,6 +52,23 @@ void SortUnique(vector<T> & v)
   v.erase(unique(v.begin(), v.end()), v.end());
 }
 
+// Sorts and removes duplicate entries from |v| according to |comp|.
+// Note. If several entries are equivalent according to |comp| an arbitrary entry of them
+// is left in |v| after a call of this method.
+// Note. |comp| should implement operator<. It means the expression
+// !comp(t1, t2) && !comp(t2, t1) should return true iff t1 is equivalent to t2.
+// It's necessary for std::unique.
+template <typename T>
+void SortUnique(function<bool(T const &, T const &)> const & comp, vector<T> & v)
```

Comment:
The vector should go first. In any case, please, pass comparer by an rvalue reference, like:
```c++
template <typename T, typename TFn>
void SortUnique(vector<T> & v, TFn && fn)
{
  ...
}
```